pkg/tests/cross-tests/tfwrite.go

```diff
@@ -56,14 +56,18 @@ func writeBlock(body *hclwrite.Body, schemas map[string]*schema.Schema, values m
 			if sch.Type == schema.TypeMap {
 				body.SetAttributeValue(key, value)
 			} else if sch.Type == schema.TypeSet {
-				for _, v := range value.AsValueSet().Values() {
-					newBlock := body.AppendNewBlock(key, nil)
-					writeBlock(newBlock.Body(), elem.Schema, v.AsValueMap())
+				if !value.IsNull() {
```

Comment:
Tolerate missing values encoded by nulls - as required by tftypes.Value (and translated to cty.Value).